File: ESAResturant.NEW.web/ClientApp/src/app/admin/item/Models/Item.ts
import { ItemPrice } from "./itemPrice";
import { ItemSize } from "./ItemSize";
import { ItemCategory } from "./ItemCategory";
import { Unit } from "./Unit";
import { BillDetail } from "../../bill-invoice/Models/BillDetail";
import { ItemComponent } from "./ItemComponent";

export class Item {

  constructor() {
    this.id = 0;
    this.code = '';
    this.nameAR = '';
    this.nameEN = '';
    this.note = '';
    this.imageName = '';
    this.imagePath = '';
    this.itemCategoryId = 0;
    this.discountId = 0;
    this.vat = 0;
    this.unitId = 0;
    this.smallUnitQty = 0;
    this.averagePrice = 0;
    this.totalAveragePrice = 0;
    this.itemPriceDefault = 0;
    //this.itemIndex = 0;

    this.isDelete = false;
    this.itemPrices = [];
  }

  id: number;
  code: string;
  nameAR: string;
  nameEN: string;
  note: string;
  imageName: string;
  imagePath: string;
  itemCategoryId: number;
  currentDicount: number;
  discountId?: number;
  vat: number;
  unitId: number;
  smallUnitQty: number;
  averagePrice: number;
  totalAveragePrice: number;
  lastPurchaseDate: Date;
  lastSeleDate: Date;
  isDelete: boolean;

  lowestSellPrice: number;
  buyPrice: number;
  lowestQuantity: number;
  isForSell: boolean;
  barCode1: string;
  barCode2: string;
  barCode3: string;
  currentQuantity: number;
  firstQuantity: number;



  itemPriceDefault?: number;
  itemIndex: number;



  itemCategory: ItemCategory;
  unit: Unit;
  itemPrices: ItemPrice[];
  billDetails: BillDetail[]
  itemComponents: ItemComponent[];
}

File: ESC.Resturant.Data/Entities/Item.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace ESC.Resturant.Data.Entities
{
    public class Item
    {
        [Key]
        public int Id { get; set; }
  
         //public Discount Discount { get; set; }
        public string Code { get; set; }
        public string NameAR { get; set; }
        public string NameEN { get; set; }
        //public int ItemPricesId { get; set; }
        public string Note { get; set; }
        public string ImageName { get; set; }
        public string ImagePath { get; set; }
        public int? ItemCategoryId { get; set; }
        public int CurrentDicount { get; set; }
        public Nullable<double> VAT { get; set; }
        public Nullable<int> UnitId { get; set; }
        public double SmallUnitQty { get; set; }
        public double AveragePrice { get; set; }
        public double TotalAveragePrice { get; set; }
        public DateTime? LastPurchaseDate { get; set; }
        public DateTime? LastSeleDate { get; set; }
        public bool? IsDelete { get; set; }
        public Nullable<decimal> LowestSellPrice { get; set; }
        public Nullable<decimal> BuyPrice { get; set; }
        public Nullable<int> LowestQuantity { get; set; }
        public Nullable<bool> IsForSell { get; set; }
        public string BarCode1 { get; set; }
        public string BarCode2 { get; set; }
        public string BarCode3 { get; set; }
        public Nullable<double> CurrentQuantity { get; set; }
        public Nullable<double> FirstQuantity { get; set; }
        public Nullable<double> ItemIndex { get; set; }


        public virtual ItemCategory ItemCategory { get; set; }
        public virtual Unit Unit { get; set; }
        public virtual IEnumerable<ItemPrices> ItemPrices { get; set; }
        public virtual ICollection<BillDetails> BillDetails { get; set; }
       public virtual IEnumerable<ItemComponent>ItemComponents { get; set; }


    }
}
